File: 백준/Silver/1904. 01타일/01타일.js
let readLine = require("readline");

let rl = readLine.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let N;
rl.on("line", (line) => {
  N = parseInt(line);
  rl.close();
});

rl.on("close", () => {
  let dp = [0, 1, 2];
  for (let i = 3; i <= N; i++) {
    dp[i] = (dp[i - 1] + dp[i - 2]) % 15746;
  }
  console.log(dp[N] % 15746);
});

File: 백준/Silver/1904. 01타일/01타일.py
N = int(input())
dp = [0] * (N+1)
dp[0] = 1
dp[1] = 1
for i in range(2,N+1):
    dp[i] = dp[i-1] + dp[i-2] % 15746
print(dp[-1] % 15746)